Libraries/Image/RCTImageView.m

```diff
@@ -152,6 +152,18 @@ - (UIImage *)image {
   return _imageView.image;
 }
 
+- (void)setDefaultImage:(UIImage *)defaultImage
+{
+  if (_defaultImage == defaultImage) {
+    return;
+  }
+  BOOL update = self.image == nil || [self.image isEqual:_defaultImage];
+  _defaultImage = defaultImage;
+  if (update) {
+    [self updateWithImage:defaultImage];
```

Comment:
I think we need to refactor the file to centralize the call `[self updateWithImage:self.image];` (we have a plenty of them in the file) inside something like `[self updateImageIfNeeded]` otherwise the approach will not work.

Reply:
I don't think that would do much good. I looked at all the call sites of `updateWithImage:` and they are all triggered under different conditions.